tests/testthat/test-unnest.R

```diff
@@ -0,0 +1,313 @@
+    as.data.frame(df) |> tidyr::unnest_longer(c("a", "b")) |> as.data.frame()
+  )
+})
+
+test_that("values_to errors with multiple columns without template", {
+  df <- polars::pl$DataFrame(
+    id = 1:2,
+    a = list(1:2, 3:4),
+    b = list(5:6, 7:8)
+  )
+
+  expect_snapshot(
+    df |> unnest_longer_polars(c(a, b), values_to = "val"),
+    error = TRUE
+  )
+})
+
+test_that("indices_to errors with multiple columns without template", {
+  df <- polars::pl$DataFrame(
+    id = 1:2,
+    a = list(1:2, 3:4),
+    b = list(5:6, 7:8)
+  )
+
+  expect_snapshot(
+    df |> unnest_longer_polars(c(a, b), indices_to = "idx"),
+    error = TRUE
+  )
+})
+
+test_that("values_to template works with multiple columns", {
+  df <- polars::pl$DataFrame(
+    x = c(1L, 2L),
+    y = list(c(1L, 2L), c(3L, 4L)),
+    z = list(c(5L, 6L), c(7L, 8L))
+  )
+
+  expect_equal(
+    df |> unnest_longer_polars(c(y, z), values_to = "{col}_val"),
+    as.data.frame(df) |>
+      tidyr::unnest_longer(
+        c(y, z),
+        values_to = "{col}_val"
+      ) |>
+      as.data.frame()
+  )
+})
+
+test_that("indices_to template works with multiple columns", {
+  df <- polars::pl$DataFrame(
+    x = c(1L, 2L),
+    y = list(c(1L, 2L), c(3L, 4L)),
+    z = list(c(5L, 6L), c(7L, 8L))
+  )
+
+  # Warning: `tidyr::unnest_longer()` places the index column immediately
+  # after the corresponding value column.
+  # `unnest_longer_polars()` currently places all index columns at the end.
+  columns <- c("x", "y", "y_idx", "z", "z_idx")
+
+  expect_equal(
+    df |>
+      unnest_longer_polars(
+        c(y, z),
+        indices_to = "{col}_idx"
+      ) |>
+      select(all_of(columns)),
+    as.data.frame(df) |>
+      tidyr::unnest_longer(
+        c(y, z),
+        indices_to = "{col}_idx"
+      ) |>
+      select(all_of(columns)) |>
+      as.data.frame()
+  )
+})
+
+test_that("both values_to and indices_to templates work together", {
+  df <- polars::pl$DataFrame(
+    x = c(1L, 2L),
+    y = list(c(1L, 2L), c(3L, 4L)),
+    z = list(c(5L, 6L), c(7L, 8L))
+  )
+
+  columns <- c("x", "y_val", "y_idx", "z_val", "z_idx")
+
+  expect_equal(
+    df |>
+      unnest_longer_polars(
+        c(y, z),
+        values_to = "{col}_val",
+        indices_to = "{col}_idx"
+      ) |>
+      select(all_of(columns)),
+    as.data.frame(df) |>
+      tidyr::unnest_longer(
+        c(y, z),
+        values_to = "{col}_val",
+        indices_to = "{col}_idx"
+      ) |>
+      select(all_of(columns)) |>
+      as.data.frame()
+  )
+})
+
+test_that("values_to template with dash separator works", {
+  df <- polars::pl$DataFrame(
+    x = c(1L, 2L),
+    y = list(c(1L, 2L), c(3L, 4L)),
+    z = list(c(5L, 6L), c(7L, 8L))
+  )
+
+  expect_equal(
+    df |> unnest_longer_polars(c(y, z), values_to = "{col}-n"),
+    as.data.frame(df) |>
+      tidyr::unnest_longer(
+        c(y, z),
+        values_to = "{col}-n"
+      ) |>
+      as.data.frame()
+  )
+})
```

Comment:
This looks similar to the previous test, is there a reason why the dash operator wouldn't work in the template? If not, we can remove this test.